cmds/mkdir/mkdir.go

```diff
@@ -16,32 +16,93 @@ package main
 import (
 	"flag"
 	"fmt"
+	"log"
 	"os"
+	"strconv"
+	"syscall"
+)
+
+type modeFlag struct {
+	set   bool
+	value string
+}
+
+func (m *modeFlag) String() string {
+	return m.value
+}
+
+func (m *modeFlag) Set(s string) error {
+	m.value = s
+	m.set = true
+	return nil
+}
+
+const (
+	cmd       = "mkdir [-m mode] [-v] [-p] <directory> [more directories]"
+	StickyBit = 01000
+	SgidBit   = 02000
+	SuidBit   = 04000
 )
 
 var (
+	mode    = &modeFlag{set: false, value: "0777"}
 	mkall   = flag.Bool("p", false, "Make all needed directories in the path")
-	mode    = flag.Int("m", 0666, "Directory mode")
 	verbose = flag.Bool("v", false, "Print each directory as it is made")
 	f       = os.Mkdir
 )
 
+func init() {
+	// Usage Definition
+	defUsage := flag.Usage
+	flag.Usage = func() {
+		os.Args[0] = cmd
+		defUsage()
+	}
+
+	// Complete Setting the flags
+	flag.Var(mode, "m", "Directory mode")
+}
+
 func main() {
 	flag.Parse()
 	if len(flag.Args()) < 1 {
-		fmt.Printf("Usage: mkdir [-m mode] [-v] [-p] <directory> [more directories]\n")
+		flag.Usage()
 		os.Exit(1)
 	}
 	if *mkall {
 		f = os.MkdirAll
 	}
+
+	// Get Correct Access Mode
+	if mode.set {
+		syscall.Umask(0)
+	}
+	accMode64bit, err := strconv.ParseUint(mode.value, 8, 32)
```

Comment:
keep the name short, per go convention, m is fine.

Reply:
Fixed